opentelemetry-sdk/src/opentelemetry/sdk/_metrics/aggregation.py

```diff
@@ -197,3 +198,57 @@ def collect(self) -> Optional[Histogram]:
             explicit_bounds=self._boundaries,
             aggregation_temporality=AggregationTemporality.DELTA,
         )
+
+
+def _convert_aggregation_temporality(
+    previous_point: Optional[_PointVarT],
+    current_point: _PointVarT,
+    aggregation_temporality: int,
+) -> _PointVarT:
+
+    current_point_type = type(current_point)
+
+    if previous_point is not None and type(previous_point) is not type(
+        current_point
+    ):
+        _logger.warning(
+            "convert_aggregation_temporality called with mismatched "
+            "point types: %s and %s",
+            type(previous_point),
+            current_point_type,
+        )
+
+        return current_point
+
+    if current_point_type is Sum:
+        if previous_point is None:
+
+            return replace(
+                current_point, aggregation_temporality=aggregation_temporality
+            )
+
+        if current_point.aggregation_temporality is aggregation_temporality:
+            return current_point
+
+        if aggregation_temporality == AggregationTemporality.DELTA:
+            value = current_point.value - previous_point.value
+
+        else:
+            value = current_point.value + previous_point.value
+
+        is_monotonic = (
+            previous_point.is_monotonic and current_point.is_monotonic
+        )
+
+        return Sum(
+            start_time_unix_nano=previous_point.start_time_unix_nano,
```

Comment:
In the case where you have an async instrument (so both previous and current points are CUMULATIVE) but the requested output temporality is DELTA, you would need `start_time_unix_nano=previous_point.time_unix_nano`

```
time --->

previous_point:
|-----------------|

current_point:
|-------------------------|

delta output:
                  |-------|
```

Reply:
Ok, changed it, please take a look, I'm not 100% sure :v: